src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -537,4 +527,8 @@ export class SmbFormComponent implements OnInit {
       }),
     );
   }
+
+  private smbNameValidator(control: AbstractControl): Observable<ValidationErrors | null> {
+    return this.smbNameValidationService.validateSmbName(control, this.existingSmbShare?.name);
```

Comment:
There's a noticeable delay before the validation is not yet complete, and the user is able to hit "Save". 

Suggestion: we could disable "Save" button to prevent the user from running into server-side validation error, and ending up with no outcome and no prefilled form to re-submit.